src/runtime_src/core/pcie/driver/linux/xocl/subdev/xgq.c

```diff
@@ -2637,6 +2645,44 @@ static struct bin_attribute bin_attr_vmr_system_dtb = {
 	.size = 0
 };
 
+static ssize_t vmr_plm_sync_read(struct file *filp, struct kobject *kobj,
+	struct bin_attribute *attr, char *buf, loff_t off, size_t count)
+{
+	struct xocl_xgq_vmr *xgq =
+		dev_get_drvdata(container_of(kobj, struct device, kobj));
+	unsigned char *blob;
+	size_t size;
+	ssize_t ret = 0;
+
+	mutex_lock(&xgq->xgq_lock);
+	blob = xgq->xgq_vmr_plm_sync;
+	size = xgq->xgq_vmr_plm_sync_size;
+	mutex_unlock(&xgq->xgq_lock);
```

Comment:
What if xgq->xgq_vmr_plm_sync and xgq->xgq_vmr_plm_sync_size changed right after this line? Is it still safe to use them at line 2667?

Reply:
good catch, we should fix this and the system_dtb sysfs. 
in real case, this is reading some logs or static device tree metadata back from VMR devices, so unlikely it will break anything, but still lets move the code below into the mutex. Say all the way too the memcpy at 2667.